verification/spec/connection-handshake/ConnectionHandshakeModule.tla

```diff
@@ -0,0 +1,231 @@
+--------------------- MODULE ConnectionHandshakeModule ---------------------
+
+EXTENDS Naturals, FiniteSets, Sequences, TLC
+
+
+CONSTANTS MaxHeight,        \* Maximum height of the local chain.
+          ConnectionIDs,    \* The set of all possible connection IDs.
+          ClientIDs,        \* The set of all possible client IDs.
+          MaxBufLen         \* Maximum length of the input and output buffers.
+
+
+ASSUME Cardinality(ConnectionIDs) >= 1
+ASSUME Cardinality(ClientIDs) >= 1
+
+
+VARIABLES
+    inBuf,      \* A buffer (Sequence) holding any message(s) incoming to the chModule.
+    outBuf,     \* A buffer (Sequence) holding outbound message(s) from the chModule.
+    store       \* The local store of the chain running this chModule. 
+
+
+ConnectionEnds ==
+    [
+        connectionID : ConnectionIDs,
+        clientID : ClientIDs
+        \* commitmentPrefix add later
+    ]
+
+
+nullConnection == [state |-> "UNINIT"]
+
+
+Heights == 1..MaxHeight
+
+
+Clients ==
+    [   
+        clientID : ClientIDs,
+        consensusState : Heights
+    ]
+
+nullClient == "no client"
+
+
+(* The set of all types which a message can take. 
+    
+    TODO: These types are also used in the Environment module.
+    Is it possible to restrict these to a single module? 
+ *)
+CHMessageTypes ==
+    { "CHMsgInit", "CHMsgTry", "CHMsgAck", "CHMsgConfirm"}
+
+
+(***************************************************************************
+ Helper operators.
+ ***************************************************************************)
+
+(* Returns true if 'para' matches the parameters in the local connection,
+    and returns false otherwise.
+ *)
+CheckLocalParameters(para) ==
+     /\ store.connection.parameters.localEnd.connectionID = para.localEnd.connectionID   
```

Comment:
This is hard to read. Maybe better define localEnd (store.connection.parameteres.localEnd with LET) and remoteEnd.